### PR TITLE
Update rubocop version and targer Ruby version

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ekohl** [comment]
Looks like the config needs to be updated

**Comment by ekohl** [comment]
```
Error: unrecognized cop Performance/Casecmp found in .rubocop.yml, unrecognized cop Performance/RedundantMatch found in .rubocop.yml, unrecognized cop Performance/RedundantMerge found in .rubocop.yml
```

**Comment by ehelms** [comment]
Yea, working on those errors, They don't show up locally >.<

**Comment by ehelms** [comment]
Updated, added some rules to skip for now.

File: katello/hooks/pre/18-vm_overcommit.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 else
   # Grab the current status of vm.overcommit from sysctl
   overcommit_check = `sysctl -n vm.overcommit_memory`.to_i
-  if overcommit_check > 0
+  if overcommit_check.positive?
```
Comment (ekohl) [inline katello/hooks/pre/18-vm_overcommit.rb]: Is this really more readable?

**Review by ekohl — APPROVED** [review]

**Pull request merged**